packages/apputils-extension/src/index.ts

```diff
@@ -297,6 +300,41 @@ const print: JupyterFrontEndPlugin<void> = {
   }
 };
 
+export const toggleHeader: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/apputils-extension:toggle-header',
+  autoStart: true,
+  requires: [ITranslator],
+  optional: [ICommandPalette],
+  activate: (
+    app: JupyterFrontEnd,
+    translator: ITranslator,
+    palette: ICommandPalette | null
+  ) => {
+    const trans = translator.load('jupyterlab');
+
+    const category: string = trans.__('Main Area');
+    app.commands.addCommand(CommandIDs.toggleHeader, {
+      label: trans.__('Show Header'),
```

Comment:
I was a little worried that since the application has a "Header" region, and people will think that this toggles the application-level header rather than the activity-level header if it is just labeled "Show Header". Is there an adjective we can put in front of "Header" that indicates it is for the widget?

On the other hand, this concern may be overblown, since we don't expose to the user the fact that the application has a top-level "header" area. It's mainly developers that would be confused, I think.

On the third hand, it may be kind of nice if something in that title directed people to watch an individual activity's content, since many of these "Show X" commands are referring to application-level things. For example "Show Activity Header" uses our notion of a thing inside of a tab is an "activity". However, on the fourth hand, we don't really use the word "Activity" in the UX much (just defined in the docs, but who reads those? :), so I wasn't really happy with "Show Activity Header" as well.

Anyways, I think whatever you want to do with this is fine - these are just some thoughts around that command name.

Reply:
Not only that, it's also a bit unnerving because, without any extensions that populate `contentHeader`, toggling it from the command palette doesn't *do* anything visually 😕.

(Five minutes later~)

I now have the following:
```ts
isEnabled: () =>
  app.shell.currentWidget instanceof MainAreaWidget &&      // this is old, from your version
  app.shell.currentWidget.contentHeader.widgets.length > 0, // this is new
```
With this in place, users won't even see that command unless toggling it will actually visually change something, which in turn makes me feel a bit more comfortable renaming the command name to *"Show Header(s) Under Toolbar"*?

What do you think?

This has the advantage of explicitly indicating what's going to change and doesn't make the "header" jargon do too much work? Is this remotely in the JupyterLab style of wording? I do agree, this seems hard since there's not a word users use to allude to the toolbar undercarriage.

Reply:
> ```ts
>   app.shell.currentWidget.contentHeader.widgets.length > 0, // this is new
> ```

> With this in place, users won't even see that command unless toggling it will actually visually change something, 

Nice! To quote you, I *love* it :). (And FYI they will *see* it, it is just greyed out as not active.)

> _Show Header(s) Under Toolbar"_?

I like how that is explicit! My suggestion with that name is that we typically don't do `(s)` as it adds complexity to the language. I think of Header as a region (which may contain multiple things), so I think a singular "Show Header Under Toolbar" is more natural -- or maybe "Show Header Above Content" to again make the connection to the content more explicit rather than the connection to the toolbar.

Reply:
> FYI they will see it, it is just greyed out as not active

Hmmm, interesting, in my JupyterLab dev instance, it doesn't appear at all until I run a command to add something to the header? E.g., without anything in `contentHeader`:

<img width="452" alt="Screen Shot 2021-03-26 at 10 43 07 PM" src="https://user-images.githubusercontent.com/37649/112711319-b517df80-8e84-11eb-9ce7-40c3870b065f.png">

> My suggestion with that name is that we typically don't do `(s)` as it adds complexity to the language

Got it! Also a pain for translators no doubt.

> "Show Header Above Content"

This sounds good 👌. I didn't know how well-established "content" was as jargon among users but I didn't realize the toolbar might be missing for some widgets so this is the correct choice.

Commit forthcoming! Thank you for your hard work!!

Reply:
> I didn't know how well-established "content" was as jargon among users

Yeah, probably not well either, but at least the colloquial meaning is clearer than "Activity"

Reply:
> Hmmm, interesting, in my JupyterLab dev instance, it doesn't appear at all until I run a command to add something to the header? E.g., without anything in contentHeader:

You're right! Looks like we have CSS that hides disabled commands:

https://github.com/jupyterlab/jupyterlab/blob/ccd6f267b49aacbfef35b5248a2be9c80a5dbbce/packages/apputils/style/commandpalette.css#L61-L65

Looks like that was a change when the command palette went modal: https://github.com/jupyterlab/jupyterlab/pull/8809/files#diff-488f8c5965a825e8e543ab3d4e6bffc297902a0ddffc6a65131677fbfb635435R64